spec.go

```diff
@@ -10,6 +10,7 @@ type Spec struct {
 	Etcd       spec.Etcd       `json:"etcd" yaml:"etcd"`
 	Kubernetes spec.Kubernetes `json:"kubernetes" yaml:"kubernetes"`
 	Masters    []spec.Node     `json:"masters" yaml:"masters"`
+	Template   spec.Template   `json:"template" yaml:"template"`
```

Comment:
This name is very generic IMO. It's next to impossible to a newcomer to discover this is a cloud config template.

Reply:
As always it would be good to provide a better alternative instead of only pointing out weaknesses. <3

I used `K8sCloudConfig` before. My reasoning was that we also use `Calico`. I do not have string feelings on this though so I just aligned it to the feedback I got as I do not mind either ways.

Reply:
> As always it would be good to provide a better alternative instead of only pointing out weaknesses. <3

Yes, first I searched for endorsement. Here's my proposal.

This template version represents guest cluster version so I think top-level field `Version` makes sense, i.e. skipping the `Template` struct. I doesn't matter for the operator's user that we version template or anything else. The user cares only about the cluster version, which created from the TPO spec. It also would give us freedom to change underlying semantics of versioning.

Or in other words I think this `Version` describes cluster, not templates from the operator user's perspective. The thing that cluster version corresponds to the k8scloudconfig template version used by the operator to create that cluster is the implementation detail we don't need to expose here.

Reply:
I don't mind either way. We also expose the implementation detail of using Calico. Whatever, when @rossf7 is fine with this I remove the `Template` block and then we only have `Version`.

Reply:
I'm fine with removing the `Template` block. I agree this is the Version of the cluster. So I like the more generic name. How the versioning is implemented is implementation detail.

Reply:
Cool. I'll do it!

I have to change `kvm-operator` and `aws-operator` for sure. Anything  else?

Reply:
Nvm. Tim did it already <3